lnwallet/channel.go

```diff
@@ -8070,26 +8303,45 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView,
 	// HTLC to the commitment, as only the balance remaining after this fee
 	// has been paid is actually available for sending.
 	feePerKw := filteredView.feePerKw
-	htlcCommitFee := lnwire.NewMSatFromSatoshis(
-		feePerKw.FeeForWeight(commitWeight + input.HTLCWeight),
+	additionalHtlcFee := lnwire.NewMSatFromSatoshis(
+		feePerKw.FeeForWeight(input.HTLCWeight),
 	)
+	commitFee := lnwire.NewMSatFromSatoshis(
+		feePerKw.FeeForWeight(commitWeight))
 
-	// If we are the channel initiator, we must to subtract this commitment
-	// fee from our available balance in order to ensure we can afford both
-	// the value of the HTLC and the additional commitment fee from adding
-	// the HTLC.
 	if lc.channelState.IsInitiator {
-		// There is an edge case where our non-zero balance is lower
-		// than the htlcCommitFee, where we could still be sending dust
-		// HTLCs, but we return 0 in this case. This is to avoid
-		// lowering our balance even further, as this takes us into a
-		// bad state where neither we nor our channel counterparty can
-		// add HTLCs.
-		if ourBalance < htlcCommitFee {
+		// When the buffer is of type `FeeBuffer` type we know we are
+		// going to send or forward an htlc over this channel therefore
+		// we account for an additional htlc output on the commitment
+		// tx.
+		futureCommitWeight := commitWeight
+		if buffer == FeeBuffer {
+			futureCommitWeight += input.HTLCWeight
+		}
+
+		// Make sure we do not overwrite `ourBalance` that's why we
+		// declare bufferAmt beforehand.
+		var bufferAmt lnwire.MilliSatoshi
+		ourBalance, bufferAmt, err = lc.applyCommitFee(
+			ourBalance, futureCommitWeight, feePerKw, buffer,
+		)
+		// TODO(ziggie): Should we add complexity here and return the
```

Comment:
what do you think about this @yyforyongyu  ?

Reply:
What about keeping this at 0, but adding a `CanSend` field for the Channel (OpenChannel I think), why bother here in returning a dust amount, if we cannot add an HTLC, we should signal 0, so that we are not draining the channel further.